server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java

```diff
@@ -159,6 +161,13 @@ && getCatalogDAO(session, updateCatalog.getNewName()) != null) {
         if (updateCatalog.getComment() != null) {
           catalogInfoDAO.setComment(updateCatalog.getComment());
         }
+        if (!updateCatalog.getProperties().isEmpty()) {
+          PropertyRepository.findProperties(session, catalogInfoDAO.getId(), Constants.CATALOG)
+              .forEach(session::remove);
+          session.flush();
+          PropertyDAO.from(updateCatalog.getProperties(), catalogInfoDAO.getId(), Constants.CATALOG)
+              .forEach(session::persist);
+        }
         catalogInfoDAO.setUpdatedAt(new Date());
         session.merge(catalogInfoDAO);
         tx.commit();
```

Comment:
The CatalogInfo returned below would not have the properties, can we use the `PropertyRepository` to add that?